src/modelfree/multi/score.py

```diff
@@ -91,6 +95,85 @@ def multi_score(score, save_path):
+        assert run.status == 'COMPLETED'
+        exp_id = run.result['exp_id']
+        output_dir[config] = exp_id
+
+    return output_dir
+
+
+def extract_data(path_generator, out_dir, experiment_dirs, ray_upload_dir):
+    """Helper method to extract data from multiple_score experiments."""
+    for experiment, experiment_dir in experiment_dirs.items():
+        experiment_root = osp.join(ray_upload_dir, experiment_dir)
+        # video_root contains one directory for each score_agent trial.
+        # These directories have names of form score-<hash>_<id_num>_<k=v>...
+        for trial_name in os.listdir(experiment_root):
+            # Each trial contains the Sacred output from score_agent.
+            # Note Ray Tune is running with a fresh working directory per trial, so Sacred
+            # output will always be at score/1.
+            trial_root = osp.join(experiment_root, trial_name)
+
+            sacred_config = osp.join(trial_root, 'data', 'sacred', 'score', '1', 'config.json')
+            with open(sacred_config, 'r') as f:
+                cfg = json.load(f)
+
+            def agent_key(agent):
+                return cfg[agent + '_type'], cfg[agent + '_path']
+
+            env_name = cfg['env_name']
+            victim_index = VICTIM_INDEX[env_name]
+            if victim_index == 0:
+                victim_type, victim_path = agent_key('agent_a')
+                opponent_type, opponent_path = agent_key('agent_b')
+            else:
+                victim_type, victim_path = agent_key('agent_b')
+                opponent_type, opponent_path = agent_key('agent_a')
+
+            if 'multicomp' in cfg['env_name']:
+                env_name = env_name_to_canonical(env_name)
+            env_name = env_name.replace('/', '-')  # sanitize
+
+            if opponent_path.startswith('/'):  # is path name
+                opponent_root = osp.sep.join(opponent_path.split(osp.sep)[:-3])
+                opponent_sacred = osp.join(opponent_root, 'sacred', 'train', '1', 'config.json')
+
+                with open(opponent_sacred, 'r') as f:
+                    opponent_cfg = json.load(f)
+
+                opponent_path = opponent_cfg['victim_path']
+
+            src_path, new_name, suffix = path_generator(trial_root=trial_root,
+                                                        env_name=env_name,
+                                                        victim_index=victim_index,
+                                                        victim_type=victim_type,
+                                                        victim_path=victim_path,
+                                                        opponent_type=opponent_type,
+                                                        opponent_path=opponent_path)
+            dst_path = osp.join(out_dir, f'{new_name}.{suffix}')
+            shutil.copy(src_path, dst_path)
+            dst_config = osp.join(out_dir, f'{new_name}.sacred')
```

Comment:
I have a minor future-confusion-avoiding dispreference for saving something that is fundamentally a .json file under the created extension .sacred

Reply:
Good point, I've renamed to `*_sacred.json`. I didn't want a JSON suffix as sometimes we're saving JSON files.